test/end2end.py

```diff
@@ -308,3 +311,14 @@ def change_env(self, cores):
         pbsnodefile.write('\n'.join(['localhost'] * cores))
         pbsnodefile.close()
         os.environ['PBS_NODEFILE'] = pbsnodefile.name
+
+
+    def test_unset_nodefile(self):
+        self.assertEqual(schedm.what_sched(False)[0], PBS)
+        nodefile = os.environ['PBS_NODEFILE']
+        del os.environ['PBS_NODEFILE']
+        # fall back to local if PBS_NODEFILE is not available
+        self.assertEqual(schedm.what_sched(False)[0], Local)
+
+        # reset nodefile for proper test cleanup
+        os.environ['PBS_NODEFILE'] = nodefile
```

Comment:
this should not be needed, get rid of `cleanup_PBS_env` instead in tearDown and just restore `self.orig_environ`